clientGQL_test.go

```diff
@@ -66,36 +71,94 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
 		autopilot.GraphQLQueryFixtureValidation(t, fmt.Sprintf("%s_request.json", endpoint)))))
 }
 
+func NewTestRequest(request string, variables string, response string) TestRequest {
+	templater := NewTestDataTemplater()
+	testRequest := TestRequest{
+		Request:   templater.ParseTemplatedString(request),
+		Variables: templater.ParseTemplatedString(variables),
+		Response:  templater.ParseTemplatedString(response),
+	}
+	if !IsValidJson(testRequest.Variables) {
+		panic(fmt.Errorf("testRequest Variables is not valid json: %s", testRequest.Variables))
+	}
+	if !IsValidJson(testRequest.Response) {
+		panic(fmt.Errorf("testRequest Response is not json: %s", testRequest.Response))
+	}
+	return testRequest
+}
+
+func NewTestDataTemplater(templateDirs ...string) *TestDataTemplater {
+	var templateFiles []string
+	for _, dir := range []string{"./testdata/templates"} {
+		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
```

Comment:
the signature of this function doesn't return an error yet we are here and the caller is hand handling an error.  Maybe we should just panic to the end user knows theres a broken template?